### PR TITLE
feat(Tooltip): Add a props to switch tooltip's appearance

### PR DESCRIPTION
**Pull request by Artikodin**
- [ ] Feature
- [ ] Fix
- [x] Enhancement

## Description
Quick change which permit to overwrite Tooltip's style.

## Todo - Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Comment by spartDev** [comment]
what do you want to override tooltip's style?

**Comment by Artikodin** [comment]
> what do you want to override tooltip's style?

By default the tooltip is dark with a white text but in that task it's white with dark text. So I wanted to overwrite the style in that specific case. 
https://tillersystems.atlassian.net/browse/TTB-1070

**Comment by spartDev** [comment]
See with arthur abour that we have a style guide. we need to choose for the tooltip color

**Comment by Artikodin** [comment]
> See with arthur abour that we have a style guide. we need to choose for the tooltip color

I already saw with Arthur and he told me there will be two style for the tooltip. One with black text on a white background ( as we can see on the ticket [TTB-1070](https://tillersystems.atlassian.net/browse/TTB-1070)  ) and a second one with white text on a black background ( as we can see on the [styleguide](https://projects.invisionapp.com/d/main#/console/14979945/316226426/preview) ).

So my question is shall I allow to overwrite the style or just create a props which change tooltip's color ?

**Comment by spartDev** [comment]
you sould use props

**Comment by Artikodin** [comment]
> you sould use props

Ok thanks I'll do that

**Comment by sun-tea** [comment]
:wave: I thought that since the style of the tooltip could vary depending of the context, and that Stardust is a generic components lib, I suggested to Artium that we could simply provide a custom style prop when needed. It seemed logic to me, but if there is a strict style guide that we want to follow for our apps, maybe Arthur can help us settle the style. So okay for an `appearance` prop if we are sure at least for now that we have a limited set of possible styles for the tooltip.

**Comment by Artikodin** [comment]
> We must have a limited set of possible styles for the tooltip! it's our style guide.
> @Artikodin @sun-tea watch this with Arthur

Yeah I already saw with Arthur and he told me they got just two appearance one dark and one white. And I'm gonna change the naming for ```appearance```. That seems to me more clear than ```invertColor```.

File: src/Tooltip/index.js [hunk 1]
```diff
@@ -30,6 +31,7 @@ class Tooltip extends React.Component {
     arrowPositionX: PropTypes.string,
     children: PropTypes.node,
     width: PropTypes.string,
+    appearance: PropTypes.oneOf(['dark', 'white']),
```
Comment (spartDev) [inline src/Tooltip/index.js]: inverse of `dark`is `light`

File: src/Tooltip/__tests__/Tooltip.test.js [hunk 2]
```diff
@@ -16,7 +16,20 @@ describe('<Tooltip />', () => {
     expect(container.firstChild).toMatchSnapshot();
   });
 
-  test('should render open a tooltip withouth a problem', () => {
+  test('should render without a problem with white appearance', () => {
```
Comment (spartDev) [inline src/Tooltip/__tests__/Tooltip.test.js]: don't use a snapshot here test properties for the light appearance with [jest styled component](https://github.com/styled-components/jest-styled-components)

**Comment by spartDev** [comment]
Last thing it is not a `refactor` but a `feature`

**Comment by Artikodin** [comment]
> use an `Object literals` helper to change the appearance. It's cleaner.
> see helpers button
> 
> ```js-jsx
> export const getAppearance = ({ palette }, appearance) => {
>   return {
>     dark: css`
>       ...
>     `,
>     light: css`
>       ...
>     `,
>   }[appearance];
> };
> ```
> 
> By the way if you want to learn more abour object literals, You can read this article: https://toddmotto.com/deprecating-the-switch-statement-for-object-literals/

Thanks for this article I didn't know about that pattern. I modify that

File: src/Tooltip/elements.js [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
 
   padding: 0.6rem 1rem;
 
   border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
 
-  color: ${({ theme: { palette } }) => palette.white};
+  font-size: 1.4rem;
 
-  background: ${({ theme: { palette } }) => palette.darkBlue};
+  /* Button Appearance: [default, primary, secondary...] */
```
Comment (spartDev) [inline src/Tooltip/elements.js]: It's not a button appearance, be careful!! about copy / past

File: src/Tooltip/__tests__/Tooltip.test.js [hunk 4]
```diff
@@ -16,7 +17,27 @@ describe('<Tooltip />', () => {
...
+        appearance="light"
+        title="Ventes nettes (ventes brutes moins les réductions et les annulations) plus les taxes
+        sur la période séléctionnée."
+      >
+        <button type="button">Show Tooltip</button>
+      </Tooltip>,
+    );
+    const tooltipNode = getByTestId('tooltip');
+
+    expect(tooltipNode).toHaveStyleRule({
```
Comment (spartDev) [inline src/Tooltip/__tests__/Tooltip.test.js]: if I check the light css I see:
```js
    color: ${palette.darkBlue};
    background: ${palette.white};
    box-shadow: 0 0 0 1px ${palette.lightGrey}, 0 2px 16px 0 rgba(0, 0, 0, 0.1);
      &::before {
        background: ${palette.white};

```
is not what you test

Comment (Artikodin) [reply src/Tooltip/__tests__/Tooltip.test.js]: Thank you for that comment. I read the documentation again and I noticed my tests was totally wrong.

File: src/Tooltip/index.js [hunk 5]
```diff
@@ -16,6 +16,7 @@ import { Container, Wrapper } from './elements';
  * @param {string} arrowPositionX // Set position of the Tooltip's arrow.
  * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment). It is the element where a tooltip is hooked to.
  * @param {string} width // Tooltip width.
+ * @param {string} appearance // Appearance is used in to set the color of the tooltip which can be dark or light.
```
[the remark this answers is not in the document]
Comment (Artikodin) [reply src/Tooltip/index.js]: yep you right

**Pull request merged**